app/models/user.rb

```diff
@@ -167,12 +176,92 @@ def self.requested_sync_approval
     where(sync_approval_status: :requested)
   end
 
-  def has_role?(*roles)
-    roles.map(&:to_sym).include?(role.to_sym)
+  def can_manage?(scope)
+    admin = Role.admin.first
+    
+    value =
+      case scope.class.model_name.name.to_sym
```

Comment:
We could consider using those convenience methods to make this something like:

```
case scope.class.model_name.name.to_sym
when :Organization
  manage_organizations.include?(scope)
when :FacilityGroup
  manage_facility_groups.include?(scope)
when :Facility
  manage_facilities.include?(scope)
end
```

Of course, this assumes that the scope is always a single record. The method could have two parts, the second for when the scope is a relation or class.